pulser/simulation.py

```diff
@@ -238,6 +238,19 @@ def build_coeffs_ops(basis, addr):
 
         self._hamiltonian = ham
 
+    def get_hamiltonian(self, time):
+        """Get the Hamiltonian created from the sequence at a fixed time.
+
+        Args:
+            time (float): The specific time in which we want to extract the
+                    Hamiltonian (in ns).
+        """
+        if time > 1000 * self._times[-1]:
+            raise ValueError("Provided time is larger than sequence duration.")
+        if time < 0:
+            raise ValueError("Provided time is negative.")
+        return self._hamiltonian(time/1000).copy()
```

Comment:
Did you check if the `copy` is necessary? It might be returning a new `QObjEvo` when you call it.

Reply:
Removed the `copy` method, since a new `Qobj` is indeed constructed from the `QobjEvo`